### PR TITLE
:rocket: Refactoring for new design, sorry for this big commit with :…

### PR DESCRIPTION
**Pull request by kosssi**
…bug: maybe...

### CODE REVIEW

File: src/app/views/file_viewer.coffee [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        @router ?= app.router
+        @loading = true
+        @files = []
+        @config ?= app.init.config
+
+        @breadcrumb = @getBreadcrumb @options.path
+
+        if @options.path isnt '/'
+            @parentPath = @options.path.replace(/\\/g,'/')
+                .replace(/\/[^\/]*$/, '')
```
Comment (nono) [inline src/app/views/file_viewer.coffee]: I think I saw some helpers for manipulating paths somewhere above in this PR

File: src/app/views/search.coffee [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        if cozyFile.docType.toLowerCase() is 'folder'
+            return 'folder'
+        else if (@mimeClasses[cozyFile.mime])
+            return @mimeClasses[cozyFile.mime]
+        else
+            return 'file'
+
+
+
+    mimeClasses:
```
Comment (nono) [inline src/app/views/search.coffee]: You have a list of mimeClasses in another view. It should probably mutualized

File: src/app/init.coffee [hunk 3]
```diff
@@ -593,11 +569,12 @@ module.exports = class Init
 
 
     loginWizard: ->
-        @app.router.navigate "login/#{@currentState}", trigger: true
+        @app.router.navigate "#onboarding/welcome", trigger: true
+#        @app.router.navigate "login/#{@currentState}", trigger: true
```
Comment (kosssi) [inline src/app/init.coffee]: this file is removed.

File: src/app/lib/first_replication.coffee [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    # 2. Put in PouchDB
+    # 2.1 : optionnaly, fetch attachments before putting in pouchDB
+    # Return the list of added doc to PouchDB.
+    _copyView: (options, callback) ->
+        log.info "enter copyView for #{options.docType}."
+        # Last step
+        putInPouch = (doc, cb) =>
+            @replicator.db.put doc, 'new_edits': false, (err) ->
+                return cb err if err
+                cb null, doc
```
Comment (kosssi) [inline src/app/lib/first_replication.coffee]: 👍

File: src/app/lib/initialize.coffee [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                            callback()
+
+
+
+
+    upsertLocalDesignDocuments: (callback) ->
+        designDocs =
+            new DesignDocuments @database.replicateDb, @database.localDb
+        unless callback
+            callback = ->
```
Comment (kosssi) [inline src/app/lib/initialize.coffee]: I add it directly on method constructor ;)

File: src/app/lib/permission.coffee [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+        @config = app.init.config
+        @permissions = cordova.plugins.permissions
+        @synchro = app.synchro
+
+
+    checkPermission: (type, success, callback) ->
+
+        if device.platform is 'iOS'
+            success true
```
Comment (kosssi) [inline src/app/lib/permission.coffee]: 😯 thanks...

File: src/app/router.coffee [hunk 7]
```diff
@@ -70,23 +166,23 @@ module.exports = class Router extends Backbone.Router
...
+#    permissions: ->
+#        app.layout.setTitle(t('setup') + ' 2/4')
+#        $('#btn-menu, #btn-back').hide()
+#        @display new PermissionsView()
+#
+#    config: ->
+#        $('#btn-back').hide()
+#        title = if app.isFirstRun then (t('setup') + ' 4/4') else t 'config'
+#        app.layout.setTitle title
+#        @display new ConfigView()
```
Comment (kosssi) [inline src/app/router.coffee]: done

File: src/app/views/search.coffee (hunk 2 above)
Comment (kosssi) [inline src/app/views/search.coffee]: now it's the reference ;)

File: src/app/views/file_viewer.coffee (hunk 1 above)
Comment (kosssi) [inline src/app/views/file_viewer.coffee]: 👍

File: src/app/lib/synchronization.coffee [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        @_canSync (err) =>
+            return callback err if err
+
+            unless @config.firstSyncIsDone()
+
+                checkFiles = (callback) =>
+                    unless @config.get 'firstSyncFiles'
+                        @firstReplication.addTask 'files', callback
+                    else
+                        callback()
```
Comment (kosssi) [inline src/app/lib/synchronization.coffee]: I change it ;)

File: src/app/lib/synchronization.coffee [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+            return callback err if err
+
+            unless @live
+                log.info 'start synchronization cozy to android'
+                @live = true if options.live
+                @replicator.startRealtime options, (err) =>
+                    @live = false if options.live
+                    callback err
+            else
+                callback()
```
Comment (kosssi) [inline src/app/lib/synchronization.coffee]: I change it ;)

File: src/app/replicator/change/change_contact_handler.coffee [hunk 10]
```diff
@@ -63,26 +64,29 @@ module.exports = class ChangeContactHandler
...
-        , cb
-        , new ContactFindOptions cozyId, false, [], AndroidAccount.TYPE, \
-            AndroidAccount.NAME
+        success = ->
+            navigator.contacts.find [navigator.contacts.fieldType.sourceId]
+            , (contacts) ->
+                callback null, contacts[0]
+            , callback
+            , new ContactFindOptions cozyId, false, [], AndroidAccount.TYPE, \
+                AndroidAccount.NAME
```
Comment (kosssi) [inline src/app/replicator/change/change_contact_handler.coffee]: ok

File: src/app/replicator/fromDevice/contact_importer.coffee [hunk 11]
```diff
@@ -17,31 +18,33 @@ module.exports = class ContactImporter
...
+                            @_delete contact, continueOnError cb
+                        else if contact.sourceId
+                            @_update contact, continueOnError cb
+                        else
+                            @_create contact, continueOnError cb
+                , callback
+
             , callback
+            , new ContactFindOptions "1", true, []
+            , AndroidAccount.TYPE, AndroidAccount.NAME
```
Comment (kosssi) [inline src/app/replicator/fromDevice/contact_importer.coffee]: 😟 it's an long time error, thanks 👍

File: src/app/migrations/2.0.0.coffee [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        newConfig = {}
+
+        replicateDb.get defaultConfig._id, (err, doc) ->
+            return callback err if err
+
+            # remove auth
+            # remove lastInitState
+            # add state
+            for key of defaultConfig
+                if doc[key] isnt undefined
```
Comment (nono) [inline src/app/migrations/2.0.0.coffee]: I'd have written:

```coffee
if key of doc
```

**Pull request merged**